### DISCUSSION TITLE
Enum and @@unique

### DISCUSSION THREAD

**Discussion by SC0d3r** [prisma/prisma#20813] [Q&A] [answered]
### Question

is there a way to use enum's certain value in @@unique
for example we have a Message model:
```js
model Message {
 id ....
 type MessageType
 user User ...
 userID String
 
 @@unique([userID, `here we use a certain value of MessageType enum`]) 
}

enum MessageType {
 SHORT
 LONG
}
```
so in the example above is there a way in db level(not application logic) limit the user to create LONG type of message only once.

Possible solution without enum is to define models for Long, Short type of message with reference to Message, Instead of defining type for Message and 
use `@@unique([userID, messageID])` in that model

but can we do it in the Message model?

`Bard` suggested to use `@@unique([userID, enum.value(MessageType, "LONG")])`, classic Bard! he drinks alot.

**Comment by ludralph** [comment, marked as answer]
Hi @SC0d3r 👋 

It is not possible to directly reference a specific value of an enum in the @unique directive in Prisma. You might need to use the possible solution you suggested to solve this.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by SC0d3r** [reply]
Thank you for the answer, Eventually I solved it without enums
